nftest/NFTestCase.py

```diff
@@ -102,41 +108,69 @@ def test(self) -> bool:
 
     def submit(self) -> sp.CompletedProcess:
         """ Submit a nextflow run """
-        nextflow_command = ["nextflow", "run", self.nf_script]
-
-        if self.profiles:
-            nextflow_command.extend(["-profile", ",".join(self.profiles)])
-
-        for config in self.nf_configs:
-            nextflow_command.extend(["-c", config])
-
-        if self.params_file:
-            nextflow_command.extend(["-params-file", self.params_file])
-
-        for param_name, path in self.reference_params:
-            nextflow_command.extend([f"--{param_name}", path])
-
-        nextflow_command.extend([
-            f"--{self.output_directory_param_name}",
-            Path(self._env.NFT_OUTPUT, self.name_for_output)
-        ])
-
-        envmod = {
-            "NXF_WORK": self.temp_dir
-        }
-
-        # Log the shell equivalent of this command
-        self._logger.info(
-            "%s %s",
-            " ".join([f"{k}={quote(v)}" for k, v in envmod.items()]),
-            sp.list2cmdline(nextflow_command)
-        )
-
-        process = popen_with_logger(
-            nextflow_command,
-            env={**os.environ, **envmod},
-            logger=self._logger
-        )
+        # Use ExitStack to handle the multiple nested context managers
+        with ExitStack() as stack:
```

Comment:
Is the other context here the subprocess context? Does that need to be tied to the stack here or does it function independently?

Reply:
Ah got it, I'm fine with keeping it as is, just wanted to make sure I wasn't missing something